package.json

```diff
@@ -785,55 +788,49 @@
                 {
                     "command": "jupyter.notebookeditor.runallcellsabove",
                     "title": "%DataScience.runAbove%",
-                    "group": "navigation@1",
+                    "group": "overflow@1",
                     "when": "notebookViewType == 'jupyter-notebook'"
                 },
                 {
                     "command": "jupyter.notebookeditor.runcellandallbelow",
                     "title": "%DataScience.runBelow%",
-                    "group": "navigation@2",
+                    "group": "overflow@2",
                     "when": "notebookViewType == 'jupyter-notebook'"
                 },
                 {
                     "command": "jupyter.notebookeditor.restartkernel",
                     "title": "%jupyter.command.jupyter.restartkernel.title%",
-                    "group": "navigation@3",
-                    "when": "notebookViewType == 'jupyter-notebook'"
+                    "group": "navigation@1",
+                    "when": "notebookViewType == 'jupyter-notebook' && jupyter.notebookeditor.canrestartNotebookkernel && !jupyter.notebookeditor.canInterruptNotebookKernel"
                 },
                 {
                     "command": "jupyter.notebookeditor.interruptkernel",
                     "title": "%jupyter.command.jupyter.interruptkernel.title%",
-                    "group": "navigation@4",
-                    "when": "notebookViewType == 'jupyter-notebook'"
+                    "group": "navigation@1",
+                    "when": "notebookViewType == 'jupyter-notebook' && jupyter.notebookeditor.canInterruptNotebookKernel"
```

Comment:
Maybe jupyter.isnotebooktrusted on interrupt / restart as well? Obviously it should be covered by canrestart or canInterrupt, but provides an extra degree of safety just in case we have a bug with those other contexts so that we don't accidentally enable something when not trusted. Surprised we didn't have the trusted check already.

Reply:
Actually they already have the enablements in commands sections? That's odd, which version wins if they are different? I'd assume the more specific one.

Reply:
The more specific one

I didn't want to disable the command in the command palette

Reply:
So is runallcellsabove enabled when not trusted then? Looks like our more specific version here for the editor doesn't have the trusted check

Reply:
It's there but disabled.